packages/i18n/src/server/createI18nHandles.ts

```diff
@@ -13,8 +13,22 @@ export const createI18nHandles = ({ supportedLocales, defaultLocale }: I18nParam
 	const defaultLanguage = localeToLanguage(defaultLocale);
 	const supportedLanguages = supportedLocales.map(localeToLanguage);
 
+	const i18nRedirect: (location: (event: RequestEvent) => string) => Handle =
+		(location) =>
+		({ event, resolve }) => {
+			if (isLocalized && event.url.pathname === '/') {
+				return new Response(null, {
+					status: 303,
+					headers: new Headers({
+						location: location(event)
+					})
+				});
+			}
+			return resolve(event);
```

Comment:
```suggestion
	const i18nRedirect: (location: (event: RequestEvent) => string) => Handle =
	
	!isLocalized ? 
	 (location) => ({ event, resolve }) => resolve(event)
	 :
		(location) =>
		({ event, resolve }) => {
			if (event.url.pathname === '/') {
				return new Response(null, {
					status: 303,
					headers: new Headers({
						location: location(event)
					})
				});
			}
			return resolve(event);
```

This way, only `({ event, resolve }) => resolve(event)` will reside in memory if not localized.